### PR TITLE
Merged PRs (23-18;16-12;10-9)

### PR DESCRIPTION
**Pull request by amadeo-alex**
***Note: following changes have been merged to https://github.com/amadeo-alex/HASS.Agent, until original development resumes new future feature requests/bugfixes will be implemented there ***

This PR servers as a point of reference with all PRs which were created by me until this point.
Due to the large amount of commits/changes in those previous PRs it will be easier to merge small changes made to the individual branches than to try and integrate 10+ PRs once Sam has time to review them.

I'll probably use this branch/PR as a starting point for new PRs.

Contains following branches:
- https://github.com/LAB02-Research/HASS.Agent.Staging/pull/23
- https://github.com/LAB02-Research/HASS.Agent.Staging/pull/22
- https://github.com/LAB02-Research/HASS.Agent.Staging/pull/21
- https://github.com/LAB02-Research/HASS.Agent.Staging/pull/20
- https://github.com/LAB02-Research/HASS.Agent.Staging/pull/19
- https://github.com/LAB02-Research/HASS.Agent.Staging/pull/18
- https://github.com/LAB02-Research/HASS.Agent.Staging/pull/16
- https://github.com/LAB02-Research/HASS.Agent.Staging/pull/15
- https://github.com/LAB02-Research/HASS.Agent.Staging/pull/14
- https://github.com/LAB02-Research/HASS.Agent.Staging/pull/13
- https://github.com/LAB02-Research/HASS.Agent.Staging/pull/12
- https://github.com/LAB02-Research/HASS.Agent.Staging/pull/10
- https://github.com/LAB02-Research/HASS.Agent.Staging/pull/9

### CODE REVIEW

**Comment by larena1** [comment]
@amadeo-alex can you release a new version of Hass Agent with all the latest changes? @LAB02-Admin is gone for months already and nobody knows if and when he will ever be back.

**Comment by amadeo-alex** [comment]
> @amadeo-alex can you release a new version of Hass Agent with all the latest changes? @LAB02-Admin is gone for months already and nobody knows if and when he will ever be back.

I'll try to get something up and running before *hopefully* Sam is back (I've had contact with him/her but recently it's been hard catch up)
Just need to think about how to approach the installation as I'm not aware of the installer repository being available anywhere.

**Comment by manonstreet** [comment]
Apologies for the cross post...  @amadeo-alex can you also merge in denisabt's awesome [screenshot sensor](https://github.com/LAB02-Research/HASS.Agent.Staging/pull/8) update into your unofficial beta?  I am looking for an easy way to compile all of your fixes/enhancements + the screenshot capability.  I am currently running the screenshot build and it's working great -- but I am keen to get your additional fixes too.

Also, related to releasing a new beta, I think even a zip file package in the interim would be helpful.  It wasn't too hard to [install manually](https://hassagent.readthedocs.io/en/latest/installation/#2-manual).

**Comment by amadeo-alex** [comment]
Evening
Regarding unofficial build please take a look here - https://github.com/hass-agent/ - it's a bit more official unofficially :D
TL;DR: More fixes/features, proper installer and an org (I've moved my repo there - the one on my account is dev only) for all stuff related to the fork

I saw your question in the screenshot sensor PR, I'll ask there if denisabt has anything against it to be transparent about anything and then integrate it if there won't be a voice of veto :)

edit @manonstreet ping for visibility

**Comment by manonstreet** [comment]
Thanks @amadeo-alex.  I'll keep an eye out for a new build with the screenshot sensor incorporated!
